r5jxrm/r5jxrm/Program.cs

```diff
@@ -0,0 +1,1111 @@
+                    Console.WriteLine($"A játékosnak volt Ásza, az értékét 1-re csökkentettük, a játékos jelenleg {jatekosSzetvalasztottKartyaErtek} van");
+                }
+                else
+                {
+                    //
+                    splitLost = true;
+                    splitting = false;
+                    i = 0;
+                }
+            }
+        }
+
+        private static void MegütésSzétválasztásNélkül(string[] paklik, bool splitLost, ref bool vesztett, ref bool isPlaying, ref bool jatekosKor, bool hasSplit, ref int kartyak, ref int jatekosKartyaErtek, string[] jatekosKartyai, ref int i)
+        {
+            if (convertToInt(paklik, kartyak) == 11 && jatekosKartyaErtek + convertToInt(paklik, kartyak) > 21)
+            {
+                //Továbbmegyünk a játékos pakliján
+                i++;
+                //Adunk egy Ász 1-es értéket a játékos paklijának
+                jatekosKartyai[i] = "A - One";
+                //Növeljük az értéket 1-el
+                jatekosKartyaErtek += 1;
+                Console.WriteLine($"Megütötted! {paklik[kartyak]} lapot kaptad, jelenleg {jatekosKartyaErtek} van");
+                //Csökkentjük a paklit
+                kartyak--;
+            }
+            //Mást üt?
+            else
+            {
+                //Továbbmegyünk a játékos pakliján
+                i++;
+                //Hozzáadjuk a kártyát
+                jatekosKartyai[i] = paklik[kartyak];
+                //Hozzáadjuk az értékét
+                jatekosKartyaErtek += convertToInt(paklik, kartyak);
+                Console.WriteLine($"Megütötted! {paklik[kartyak]} lapot kaptad, jelenleg {jatekosKartyaErtek} van");
+                //Csökkentjük a paklit
+                kartyak--;
+            }
+            //Ha a játékos átlépte a 21-et
+            if (jatekosKartyaErtek > 21)
+            {
+                //Megnézzük van-e Ász 11-es értékkel
+                int index = checkHaVanAsz(jatekosKartyai);
+                if (jatekosKartyai[index] == "A")
+                {
+                    jatekosKartyai[index] = "A - One";
+                    jatekosKartyaErtek -= 10;
+                    Console.WriteLine($"Volt 11-es Ász, az értéke megváltozott 1-re, jelenleg {jatekosKartyaErtek} van");
+                }
+                else
+                {
+                    vesztett = true;
+                    if (!hasSplit || hasSplit && splitLost)
+                    {
+                        isPlaying = false;
+                    }
+                    else
+                    {
+                        jatekosKor = false;
+                    }
+                }
+            }
+        }
+
+        private static void DuplazasPluszMegütes(string[] paklik, ref double tet, double ennyivan, ref bool errorSave, ref bool vesztett, ref bool isPlaying, ref bool jatekosKor, ref int kartyak, ref int jatekosKartyaErtek, string[] jatekosKartyai, ref int i)
```

Comment:
jáájj, ha ennyi referencia átpasszolásod van, akkor valószínű egy állapotgéped van, aminek az állapotát módosítod. Erre van design pattern, state-nek hívják, de ha nem is ilyen irányba mennél el a kóddal, akkor is érdemes lehet egy központi `State` objektumba szervezni azt, amit referenciaként adsz át most.